### PR TITLE
Mock shop df tutorial

### PR DESCRIPTION
**Pull request by mrlaessig**
## Purpose

This PR includes the mock.shop Delivery function tutorial and the corresponding code.

### CODE REVIEW

File: examples/hosted-delivery-function-mock-shop/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This repo contains to the code for the mock.shop delivery function tutorial.
+
+## How to use
+
+Execute create-contentful-app with npm, npx or yarn to bootstrap the example:
+
+```bash
+# npx
+npx create-contentful-app --typescript
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/README.md:11]:
```suggestion
npx create-contentful-app --example hosted-delivery-function-mock-shop
```

File: examples/hosted-delivery-function-mock-shop/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+# npm
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/README.md:13]:
```suggestion

```

File: examples/hosted-delivery-function-mock-shop/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+npm init contentful-app -- --typescript
+
+# Yarn
+yarn create contentful-app --typescript
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/README.md:17]:
```suggestion

```

File: examples/hosted-delivery-function-mock-shop/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Execute create-contentful-app with npm, npx or yarn to bootstrap the example:
+
+```bash
+# npx
+npx create-contentful-app --typescript
+
+# npm
+npm init contentful-app -- --typescript
+
+# Yarn
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/README.md:16]:
```suggestion

```

File: examples/hosted-delivery-function-mock-shop/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+## How to use
+
+Execute create-contentful-app with npm, npx or yarn to bootstrap the example:
+
+```bash
+# npx
+npx create-contentful-app --typescript
+
+# npm
+npm init contentful-app -- --typescript
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/README.md:14]:
```suggestion

```

File: examples/hosted-delivery-function-mock-shop/src/locations/Home.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { Heading } from '@contentful/f36-components';
```
Comment (kdamball) [inline examples/hosted-delivery-function-mock-shop/src/locations/Home.tsx:1]: Why is this file required? Wondering if we need to use the home location at all

**Review by kdamball — APPROVED** [review]

File: examples/hosted-delivery-function-mock-shop/delivery-functions/mockShop.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      contentTypeId,
+      fieldId: field.id,
+      graphQLOutputType: 'Product',
+      graphQLQueryField: 'product',
+      graphQLQueryArguments: { id: '' },
+    };
+  });
+
+  return {
+    namespace: 'MockShopTutorial',
```
Comment (BasKiers) [inline examples/hosted-delivery-function-mock-shop/delivery-functions/mockShop.ts:15]: Nit. personally I think it would be more clean to just use `MockShop`

**Review by BasKiers — APPROVED** [review]

File: examples/hosted-delivery-function-mock-shop/src/locations/ConfigScreen.tsx
Comment (BasKiers) [inline examples/hosted-delivery-function-mock-shop/src/locations/ConfigScreen.tsx:1]: Since we are not going in-depth on how this file works in the tutorial it would be very helpful to add in-line comments to try to help customers understand what is happening and why (especially with the many React Hooks that are used). Preferably I would also like to see the other React components have some more inline documentation on what does what.

**Review by cemreyuksel — APPROVED** [review]
🎉
